### DISCUSSION TITLE
Slack notification config advice

### DISCUSSION THREAD

**Discussion by UNiXMIT** [semaphoreui/semaphore#1858] [Q&A]
I see the option to enable Slack notifications and set the url in the config file. I also see the option in the project settings to enable alerts. I also see the option to enable alerts in the users settings. Are all these needed? What if a user disabled alerts in their user settings, will I no longer be alerted when tasks they run fail? I also don't see a way, as an admin to modify a users 'send alerts' option and I can't login as them to change it. Thanks for any advice. Getting a bit confused with all the options for notifications. Thanks

**Comment by fiftin** [comment]
Agree with you :)

**Reply by UNiXMIT** [reply]
I still can't get Slack notifications to work in 2.9.56 and I have the 2 config settings set and alerts checkboxes set and alerts aren't set to be suppressed in the task. Is it supposed to be fixed? Could I have missed a step? Thanks

**Reply by UNiXMIT** [reply]
Helpful message ha ha
![image](https://github.com/ansible-semaphore/semaphore/assets/4117073/0c0c6f50-8e17-4e73-bd63-c0397d6b181d)
So it seems to be an issue with my Discord webhook with /slack appended to it - https://birdie0.github.io/discord-webhooks-guide/other/slack_formatting.html
Although the Discord webhook works with slack formatted requests it doesn't seem to like something coming from Semaphore.
With a real Slack webhook it works.

**Reply by UNiXMIT** [reply]
@fiftin Are there any plans to support Discord webhooks or at least a format that the Discord webhooks with slack formatting support? Thanks

**Reply by fiftin** [reply]
We need to move messenger configs to database before adding new messengers. We already have a lot of options in `config.json`.

**Reply by masterwishx** [reply]
Also interesting in discord , also /slack not working with discord webhook

**Comment by jamesholland-uk** [comment]
FWIW, I found I needed the two config file options, then the per-project alerts setting. Per-user alert settings did not seem to matter. I sending to native Slack, not Discord.

I still don't know what the per-user alert setting does though, it does not seem to be documented.
